javascript/test/wombatUrlRewriting.js

```diff
@@ -0,0 +1,852 @@
+
+test.skip('contentWithEncodedHyphen1', (t) => {
+  t.is(
+    urlRewriteFunction(
+      t.context.currentUrl,
+      t.context.originalHost,
+      t.context.originalScheme,
+      t.context.originalUrl,
+      t.context.prefix,
+      'https://www.example.com/javascript/cont%2Dnt.txt', // - is unreserved, it must be not be encoded
+      undefined,
+      undefined,
+      undefined,
+    ),
+    'http://library.kiwix.org/content/myzim_yyyy-mm/www.example.com/javascript/cont-nt.txt',
+  );
+});
+
+test.skip('contentWithEncodedHyphen2', (t) => {
+  t.is(
+    urlRewriteFunction(
+      t.context.currentUrl,
+      t.context.originalHost,
+      t.context.originalScheme,
+      t.context.originalUrl,
+      t.context.prefix,
+      'https://www.example.com/javascript/co%25nt%2Dnt.txt', // - is unreserved, it must be not be encoded
+      undefined,
+      undefined,
+      undefined,
+    ),
+    'http://library.kiwix.org/content/myzim_yyyy-mm/www.example.com/javascript/co%25nt-nt.txt',
+  );
+});
+
+test.skip('contentWithEncodedUnderscore', (t) => {
+  t.is(
+    urlRewriteFunction(
+      t.context.currentUrl,
+      t.context.originalHost,
+      t.context.originalScheme,
+      t.context.originalUrl,
+      t.context.prefix,
+      'https://www.example.com/javascript/cont%5Fnt.txt', // _ is unreserved, it must be not be encoded
+      undefined,
+      undefined,
+      undefined,
+    ),
+    'http://library.kiwix.org/content/myzim_yyyy-mm/www.example.com/javascript/cont_nt.txt',
+  );
+});
+
+test.skip('contentWithEncodedPeriod', (t) => {
+  t.is(
+    urlRewriteFunction(
+      t.context.currentUrl,
+      t.context.originalHost,
+      t.context.originalScheme,
+      t.context.originalUrl,
+      t.context.prefix,
+      'https://www.example.com/javascript/content%2Etxt', // . is unreserved, it must be not be encoded
+      undefined,
+      undefined,
+      undefined,
+    ),
+    'http://library.kiwix.org/content/myzim_yyyy-mm/www.example.com/javascript/content.txt',
+  );
+});
+
+test('contentWithSimpleQueryString', (t) => {
+  t.is(
+    urlRewriteFunction(
+      t.context.currentUrl,
+      t.context.originalHost,
+      t.context.originalScheme,
+      t.context.originalUrl,
+      t.context.prefix,
+      'https://www.example.com/javascript/content.txt?query=value',
+      undefined,
+      undefined,
+      undefined,
+    ),
+    'http://library.kiwix.org/content/myzim_yyyy-mm/www.example.com/javascript/content.txt%3Fquery%3Dvalue',
+  );
+});
+
+test.skip('contentWithQueryValueEqualSign', (t) => {
+  t.is(
+    urlRewriteFunction(
+      t.context.currentUrl,
+      t.context.originalHost,
+      t.context.originalScheme,
+      t.context.originalUrl,
+      t.context.prefix,
+      'https://www.example.com/javascript/content.txt?query=val%3Deue',
+      undefined,
+      undefined,
+      undefined,
+    ),
+    'http://library.kiwix.org/content/myzim_yyyy-mm/www.example.com/javascript/content.txt%3Fquery%3Dval%3Deue',
```

Comment:
Should we encode the original `%` (in `%3D`) ?
Else when server will decode the request path, it will decode it to `https://www.example.com/javascript/content.txt?query=val=eue` while the website is searching for `https://www.example.com/javascript/content.txt?query=val%3Deue`

Reply:
In python we are not encoding the original `%`, we keep it in fully decoded form inside the ZIM path. I've done the same in Javascript.